querybook/webapp/components/QueryViewNavigator/QueryViewNavigator.tsx

```diff
@@ -19,148 +20,138 @@ import { QueryViewFilter } from './QueryViewFilter';
-        const { isLoadingQueries, loadQueries } = this.props;
-
-        if (!isLoadingQueries && this.navigatorScrollRef) {
-            const el = this.navigatorScrollRef.current;
-            if (el.scrollHeight - el.scrollTop === el.clientHeight) {
-                // Scrolled to bottom
-                loadQueries();
+        },
+        [pollQueryExecution]
+    );
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const loadMoreIfScrolledToBottom = useCallback(
+        debounce(() => {
+            if (!isLoadingQueries && navigatorScrollRef) {
+                const el = navigatorScrollRef.current;
+                if (el.scrollHeight - el.scrollTop === el.clientHeight) {
+                    // Scrolled to bottom
+                    loadQueries();
+                }
             }
-        }
-    }
-
-    public componentDidMount() {
-        this.props.initializeFromQueryParam();
-        this.setupPolling(this.props.queryResults);
-    }
-
-    public componentDidUpdate(prevProps) {
-        if (this.props.queryResults !== prevProps.queryResults) {
-            this.setupPolling(this.props.queryResults);
-        }
-    }
-
-    public render() {
-        const {
-            queryResults,
-            isLoadingQueries,
-            queryViewFilters,
-            queryEngines,
-            queryEngineById,
-
-            initializeFromQueryParam,
-            updateFilter,
-        } = this.props;
-
-        const queryViewFilterDOM = (
-            <QueryViewFilter
-                filters={queryViewFilters}
-                updateFilter={updateFilter}
-                onRefresh={initializeFromQueryParam}
-                queryEngines={queryEngines}
-                queryEngineById={queryEngineById}
-            />
-        );
-
-        const queryResultsListDOM = queryResults.map((queryResult) => (
-            <QueryResult
-                key={queryResult.id}
-                queryExecution={queryResult}
-                queryEngineById={queryEngineById}
-            />
-        ));
-
-        const loadingDOM = isLoadingQueries ? (
-            <div className="flex-column m24">
-                <Icon name="Loading" className="mb16" />
-                <AccentText color="light" weight="bold">
-                    Loading Executions
-                </AccentText>
-            </div>
+        }, 500),
+        [loadQueries, isLoadingQueries]
+    );
+
+    const handleNavigatorScroll = useCallback(
+        (event) => {
+            if (event.target === navigatorScrollRef.current) {
+                loadMoreIfScrolledToBottom();
+            }
+        },
+        [loadMoreIfScrolledToBottom]
+    );
+
+    useEffect(() => {
+        initializeFromQueryParam();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    useEffect(() => {
+        setupPolling(queryResults);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [queryResults]);
```

Comment:
why does it not need to depend on setupPolling?

Reply:
We only need to poll when the queries change, when setupPoll change we don't need to do the polling again